packages/kbn-config-schema/src/types/object_type.ts

```diff
@@ -181,6 +181,25 @@ export class ObjectType<P extends Props = any> extends Type<ObjectResultType<P>>
     return new ObjectType(extendedProps, extendedOptions);
   }
 
+  public extendsDeep(options: ExtendsDeepOptions) {
+    const extendedProps = Object.entries(this.props).reduce((memo, [key, value]) => {
+      if (value !== null && value !== undefined) {
+        return {
```

Comment:
Good point! I updated the description and tests (https://github.com/elastic/kibana/pull/156214/commits/fc31e3285c1fcefa3d94b0892204b54b5831c0c8) so the bad way to do it can remain our little secret.